### PR TITLE
Default run_options for Primitives

### PR DESCRIPTION
**Pull request by ikkoham**
### Summary

Add `run_options` arguments to constructor of primitives and `run_options` methods to  primitives. Users can set default ``run_options`` in addition to passing  `run_options` at runtime.

### Details and comments

### CODE REVIEW

File: qiskit/primitives/base_sampler.py [hunk 1]
```diff
@@ -209,6 +215,27 @@ def parameters(self) -> tuple[ParameterView, ...]:
...
+        """Set options values for the estimator.
+
+        Args:
+            **fields: The fields to update the options
+
+        Returns:
+            self
+        """
+        self._run_options.update_options(**fields)
+        return self
```
Comment (t-imamichi) [inline qiskit/primitives/base_sampler.py:237]: Does setter need to return `self`? Do you think of a chain operation `sampler.set_run_options(...).run(...)`?

Comment (ikkoham) [reply qiskit/primitives/base_sampler.py:237]: Thanks... i didn't think so deeply...
I'll remove it if there is no use case.

File: qiskit/primitives/base_estimator.py [hunk 2]
```diff
@@ -145,6 +147,7 @@ def __init__(
                 will be bound. Defaults to ``[circ.parameters for circ in circuits]``
                 The indexing is such that ``parameters[i, j]`` is the j-th formal parameter of
                 ``circuits[i]``.
+            run_options: runtime options.
```
Comment (Cryoris) [inline qiskit/primitives/base_estimator.py:150]: Could we add a bit more information here, like "The runtime options for individual estimator executions, such as the number of shots." ? 🙂

Comment (ikkoham) [reply qiskit/primitives/base_estimator.py:150]: Thanks. Whether number of shots is supported or not is implementation-dependent, so it is not known at the base class level and cannot be written here. 

> The runtime options for individual estimator executions.

is fine.

**Comment by ikkoham** [comment]
Wait for the Runtime side to design the Option.

File: qiskit/primitives/base_sampler.py (hunk 1 above)
Comment (Cryoris) [reply qiskit/primitives/base_sampler.py:237]: I would also not expect this to return anything as it's called "set_..." 🙂

**Review by Cryoris — COMMENTED** [review]
One minor comment on the return value of `set_run_options` but otherwise this LGTM! The linting seems to fail due to a missing black formatting 🙂

**Review by Cryoris — APPROVED** [review]
LGTM, thanks for the update!

**Pull request merged**